server/src/test/java/org/elasticsearch/index/engine/InternalEngineTests.java

```diff
@@ -3638,7 +3640,8 @@ public void testRetryWithAutogeneratedIdWorksAndNoDuplicateDocs() throws IOExcep
         long autoGeneratedIdTimestamp = 0;
 
         Engine.Index index = new Engine.Index(newUid(doc), doc, UNASSIGNED_SEQ_NO, 0,
-            Versions.MATCH_ANY, VersionType.INTERNAL, PRIMARY, System.nanoTime(), autoGeneratedIdTimestamp, isRetry, UNASSIGNED_SEQ_NO, 0);
+            Versions.MATCH_DELETED, VersionType.INTERNAL, PRIMARY, System.nanoTime(), autoGeneratedIdTimestamp, isRetry,
```

Comment:
I think both `MATCH_ANY` and `MATCH_DELETED` should work? If so, I suggest to pick one of them randomly.